ClinGen/clincoded
# Connects to #914 kl progress bar style

This PR modifies layout of progress bar and does not change functionality of pathogenicity calculator. 

**Compared to the layout in R7alpha2, changes in this PR includes:**
 1. Changed background color of progress bar to light grey.
 2. Three groups (Benign criteria, Pathogenic criteria and Calculated Pathogenicity) are displayed horizontally in large, medium and small browser sizes, but stacked vertically in extra small size. 
 3. Removed border of each group.
 4. Enlarge white space between adjacent groups so that they can be visually easy to be separated.
 5. Number of criteria met at each strength level is shown in a badge and bind to relevant strength label as one element. When wrapped, the whole element will be moved to a new line so that there is no break between strength label and its number.
 6. Add icon at left of each group.

**Important Code changes**
- In file `src/clincoded/static/components/variant_central/interpretation/shared/calculator.js`:
  - Delete class `TestCalculator`. This class was temporarily used for testing and is not necessary any longer.
  - Relocate function `calculatePathogenicity` as a internal function in class `PathogenicityCalculator`.
  - Edit function `render` and use Bootstrap grid system to handle layout change.

- In file `src/clincoded/static/scss/clincoded/modules/_curator.scss`:
  - Add new css classes for layout elements.

- In file `src/clincoded/static/scss/fontawesome/_icons.scss`:
  - Use calculator icon.

**Steps to Test the Layout**
- Set browser to *LARGE SIZE* (> 1200 pixels in width).
- In VCI, enter ClinVar variation id "12345", click `Save and View Evidence`, and then click `Start a New Interpretation` button. Expect to see the progress bra as:
![screen shot 2016-09-08 at 4 25 32 pm](https://cloud.githubusercontent.com/assets/9206696/18370301/eba7d6c0-75e0-11e6-89b7-b01bb08704b8.png)
- Associate disease "ORPHA777".
- In Population tab, change `BA1` to 'Met' and then click `Save`.
- In Predictors/Missense, change `BP4` to `Met`, change `BP1` to `BP1_Strong` and click `Save`.
- In Predictors/Loss of Function, change `PVS1` to `Met` and click `Save`.
- In Segregation/Case, change `PS4` to `Met` and click `Save`, change `PP1` to `Met` and click `Save`, change `PM6` to `Met` and click `Save`.
- After selections above, expect to see progress bar (in *LARGE SIZE* browser) as:
![screen shot 2016-09-08 at 6 13 53 pm](https://cloud.githubusercontent.com/assets/9206696/18372259/13e157ec-75f0-11e6-8eaf-d94db5530b79.png)
- Narrow browser to *MEDIUM SIZE* (the first change of layout), expect to see the progress bar as:
![screen shot 2016-09-08 at 6 11 23 pm](https://cloud.githubusercontent.com/assets/9206696/18372221/b7b48be2-75ef-11e6-94fe-17f1bfe4f316.png)

- Narrow browser to *SMALL SIZE* (the second change of layout), expect to see the progress bar as:
![screen shot 2016-09-08 at 6 12 10 pm](https://cloud.githubusercontent.com/assets/9206696/18372234/ce25da84-75ef-11e6-8386-0cf5cfdbe70c.png)

- Narrow browser to extra small size (the third change of layout), expect to see the progress bar as:
![screen shot 2016-09-08 at 6 15 26 pm](https://cloud.githubusercontent.com/assets/9206696/18372286/44d9bc22-75f0-11e6-811a-a81d3ba3d22b.png)

- Enlarge browser back to *SMALL SIZE*, *MEDIUM SIZE* and then *LARGE SIZE*. Expect to see relevant layout change as above each.

**End of Test**
